CRM/Event/Form/ParticipantFeeSelection.php

```diff
@@ -310,6 +309,71 @@ protected function buildAmount($form, $discountId, $priceSetID) {
     $form->assign('priceSet', $form->_priceSet);
   }
 
+  /**
+   *
+   */
+  private function formatFieldsForOptionFull(): void {
+    $form = $this;
+    $priceSet = $form->get('priceSet');
+    $priceSetId = $form->get('priceSetId');
+    $defaultPricefieldIds = [];
+    if (!empty($form->_values['line_items'])) {
+      foreach ($form->_values['line_items'] as $lineItem) {
+        $defaultPricefieldIds[] = $lineItem['price_field_value_id'];
+      }
+    }
+    if (!$priceSetId ||
+      !is_array($priceSet) ||
+      empty($priceSet) || empty($priceSet['optionsMaxValueTotal'])
+    ) {
+      return;
+    }
+
+    //get the current price event price set options count.
+    $currentOptionsCount = CRM_Event_Form_Registration::getPriceSetOptionCount($form);
+    $recordedOptionsCount = CRM_Event_BAO_Participant::priceSetOptionsCount($form->_eventId, []);
+    $optionFullTotalAmount = 0;
+    $currentParticipantNo = (int) substr($form->_name, 12);
+    foreach ($form->_feeBlock as & $field) {
```

Comment:
@eileenmcnaughton isn't this now a dynamic property of this class?

Reply:
no more than it was before... it is already causing ugliness on that front - but by moving it back to the class then we can address